plugin/gcp/instance.py

```diff
@@ -0,0 +1,239 @@
+# #######
+# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+from plugin.gcp import utils
+from plugin.gcp.service import GoogleCloudPlatform
+from plugin.gcp.service import GCPError
+from plugin.gcp.service import blocking
+
+
+class Instance(GoogleCloudPlatform):
+    ACCESS_CONFIG = 'External NAT'
+    ACCESS_CONFIG_TYPE = 'ONE_TO_ONE_NAT'
+    NETWORK_INTERFACE = 'nic0'
+    STANDARD_MACHINE_TYPE = 'n1-standard-1'
+
+    def __init__(self,
+                 config,
+                 logger,
+                 instance_name,
+                 image=None,
+                 machine_type=None,
+                 startup_script=None,
+                 external_ip=False,
+                 tags=[]):
+        """
+        Create Instance object
+
+        :param config: gcp auth file
+        :param logger: logger object
+        :param instance_name: name of the instance
+        :param image: image id in Google Cloud Platform
+        :param machine_type: machine type on GCP, default None
+        :param startup_script: shell script text to be run on instance startup,
+        default None
+        :param external_ip: boolean external ip indicator, default False
+        :param tags: tags for the instance, default []
+        """
+        super(Instance, self).__init__(config, logger)
+        self.project = config['project']
+        self.zone = config['zone']
+        self.name = utils.get_gcp_resource_name(instance_name)
+        self.image = image
+        self.machine_type = machine_type \
+            if machine_type else self.STANDARD_MACHINE_TYPE
```

Comment:
if machine type is missing - we should probably fail.. (you defined a default value for that in the plugin.yaml, which is good because the user can see and change it there)